library/src/scala/internal/quoted/Matcher.scala

```diff
@@ -175,8 +175,8 @@ private[quoted] object Matcher {
           case (Literal(constant1), Literal(constant2)) if constant1 == constant2 =>
             matched
 
-          case (Typed(expr1, tpt1), Typed(expr2, tpt2)) =>
-            expr1 =?= expr2 && tpt1 =?= tpt2
+          case (Typed(expr1, _), pattern) =>
+            expr1 =?= pattern
```

Comment:
This means there is no way to match against a typed expression. It's fine in practice. As this is a design choice, it's better to add a small doc here, and add a link to the relevant issue.

Reply:
That should be added to the docs in #8687. These do not exist yet.